CI: api.test_api.TestPDApi.test_api fails irrelevantly

#### Problem description

In the pull request #26940, I have changed a piece of code in `pandas/_libs/window.pyx`. Tests pass successfully (including those in tests/api) on my local but not on CI. It seems that `hard_dependencies, dependency, missing_dependencies` are on the pd namespace and not deleted as supposed (see [here](https://github.com/pandas-dev/pandas/blob/master/pandas/__init__.py#L17)). But I don't think changes in #26940 cause this assertion error.

#### Error ([Here](https://travis-ci.org/pandas-dev/pandas/jobs/547916861#L2443))
```python
=================================== FAILURES ===================================
______________________________ TestPDApi.test_api ______________________________
[gw1] linux -- Python 3.7.3 /home/travis/miniconda3/envs/pandas-dev/bin/python
self = <pandas.tests.api.test_api.TestPDApi object at 0x7f2145f522e8>
    def test_api(self):
    
        self.check(pd,
                   self.lib + self.misc +
                   self.modules + self.deprecated_modules +
                   self.classes + self.deprecated_classes +
                   self.deprecated_classes_in_future +
                   self.funcs + self.funcs_option +
                   self.funcs_read + self.funcs_to +
                   self.deprecated_funcs_in_future +
                   self.deprecated_funcs + self.private_modules,
>                  self.ignored)
pandas/tests/api/test_api.py:113: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/tests/api/test_api.py:18: in check
    tm.assert_almost_equal(result, expected)
pandas/util/testing.py:325: in assert_almost_equal
    **kwargs)
pandas/_libs/testing.pyx:65: in pandas._libs.testing.assert_almost_equal
    cpdef assert_almost_equal(a, b,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
>   raise_assert_detail(obj, '{0} length are different'.format(obj),
E   AssertionError: Iterable are different
E   
E   Iterable length are different
E   [left]:  132
E   [right]: 129
E   [diff]: ['dependency', 'missing_dependencies', 'hard_dependencies']
pandas/_libs/testing.pyx:163: AssertionError
```

--- a/pandas/tests/test_downstream.py
+++ b/pandas/tests/test_downstream.py
@@ -133,7 +133,7 @@ def test_pyarrow(df):
     tm.assert_frame_equal(result, df)
 
 
-def test_missing_required_dependency(capsys):
+def test_missing_required_dependency():
     # GH 23868
     # use the -S flag to disable site-packages
     call = ['python', '-S', '-c', 'import pandas']

--- a/pandas/tests/test_downstream.py
+++ b/pandas/tests/test_downstream.py
@@ -1,7 +1,6 @@
 """
 Testing that we work in the downstream packages
 """
-import builtins
 import importlib
 import subprocess
 import sys
@@ -134,30 +133,13 @@ def test_pyarrow(df):
     tm.assert_frame_equal(result, df)
 
 
-def test_missing_required_dependency(monkeypatch):
+def test_missing_required_dependency(capsys):
     # GH 23868
-    original_import = __import__
+    # use the -S flag to disable site-packages
+    call = ['python', '-S', '-c', 'import pandas']
 
-    def mock_import_fail(name, *args, **kwargs):
-        if name == "numpy":
-            raise ImportError("cannot import name numpy")
-        elif name == "pytz":
-            raise ImportError("cannot import name some_dependency")
-        elif name == "dateutil":
-            raise ImportError("cannot import name some_other_dependency")
-        else:
-            return original_import(name, *args, **kwargs)
+    with pytest.raises(subprocess.CalledProcessError) as exc:
+        subprocess.check_output(call, stderr=subprocess.STDOUT)
 
-    expected_msg = (
-        "Unable to import required dependencies:"
-        "\nnumpy: cannot import name numpy"
-        "\npytz: cannot import name some_dependency"
-        "\ndateutil: cannot import name some_other_dependency"
-    )
-
-    import pandas as pd
-
-    with monkeypatch.context() as m:
-        m.setattr(builtins, "__import__", mock_import_fail)
-        with pytest.raises(ImportError, match=expected_msg):
-            importlib.reload(pd)
+    output = exc.value.stdout.decode()
+    assert all(x in output for x in ['numpy', 'pytz', 'dateutil'])

